tests/attestation_plugin/plugin/tests.c

```diff
@@ -786,6 +789,150 @@ static void _test_time_policy(
         OE_VERIFY_FAILED_TO_FIND_VALIDITY_PERIOD);
 }
 
+static void _buffer_to_hex(
+    char* hex_string,
+    size_t hex_string_size,
+    const uint8_t* buffer,
+    const size_t buffer_size)
+{
+    OE_TEST(hex_string_size >= buffer_size * 2);
+
+    char* conversion_str = "0123456789ABCDEF";
+
+    for (size_t i = 0; i < buffer_size; i++)
+    {
+        hex_string[2 * i] = conversion_str[(buffer[i] >> 4)];
+        hex_string[2 * i + 1] = conversion_str[(buffer[i] & 0x0f)];
+    }
+}
+
+static void _convert_raw_to_hex_data(
```

Comment:
`_convert_raw_data_to_hex_data` or `_convert_byte_to_hex_data`?
Actually the conversion part is entirely inside of `_buffer_to_hex`. Maybe you can separate another function like `bool is_hex_data()` to verify the hex string. If true, you can convert the buffer using `_buffer_to_hex` directly.